torch/_torch_docs.py

```diff
@@ -1209,6 +1209,42 @@ def merge_dicts(*dicts):
 
 """.format(**common_args))
 
+add_docstr(torch.tensor_split,
+           r"""
+tensor_split(input, indices_or_sections, dim=0) -> List of Tensors
+
+Split a tensor into multiple sub-tensors, all of which are views of the
+input tensor. This function is based on :func:`numpy.array_split`.
+
+Args:
+    input (Tensor): the tensor to split
+    indices_or_sections (int or (list(int))):
+        If :attr:`indices_or_sections` is an integer, it represents the number of
+        sections to divide the input into. The input does not have to be divisible
+        by the number of sections. The first :code:`int(input.size(dim) % sections)`
+        output tensors will have have size :code:`int(input.size(dim) / sections) + 1`
+        in dimension :attr:`dim`. The rest will have size
+        :code:`int(input.size(dim) / sections)`.
+
+        If :attr:`indices_or_sections` is a list of ints, it represents a list of
+        indices where the input will be split. For instance, :code:`[2, 3]` and
+        :code:`dim=0` would result in the following tensors:
+
+            - :code:`input[:2]`
+            - :code:`input[2:3]`
+            - :code:`input[3:]`
+
+    dim (int, optional): dimension along which to split the tensor. Default: ``0``
+
+Example::
+    >>> x = torch.arange(8)
+    >>> torch.tensor_split(x, 3)
+    (tensor([0, 1, 2]), tensor([3, 4, 5]), tensor([6, 7]))
+    >>> x = torch.arange(7)
```

Comment:
Also good. This needs an example where a list of integers is provided, and an example where dim != 0.

Reply:
Good idea, added.